services/companyService.go

```diff
@@ -28,3 +29,29 @@ func (service *CompanyService) UpdateCompany(updateCompany models.Company) error
 func (service *CompanyService) DeleteCompany(id string) error {
 	return service.Repository.DeleteCompany(id)
 }
+
+func (service *CompanyService) ApproveExternalAccess(companyID string, idear string) error {
```

Comment:
ApproveExternalAccess and DissaproveExternalAccess are basically the same method, with one bool parameter which could be passed, true for approving, false otherwise. Also this could be writen like (company, idear string)

Reply:
We left two method in controller becouse we have two paths **/approve/{idear}** and **/disapprove/{idear}** and made one method in service and repo (ChangeExternalRightApproveStatus) that accept additional bool parameter